client/controller/index.web.js

```diff
@@ -219,6 +224,25 @@ export function redirectWithoutLocaleParamIfLoggedIn( context, next ) {
 	next();
 }
 
+/**
+ * Removes the locale parameter from the beginning of the path, and redirects logged-in users to it.
+ *
+ * @param   {Object}   context Context object
+ * @param   {Function} next    Calls next middleware
+ * @returns {void}
+ */
+export const redirectWithoutLocaleParamInFrontIfLoggedIn = ( context, next ) => {
+	if ( isUserLoggedIn( context.store.getState() ) ) {
+		const pathWithoutLocale = removeLocaleFromPathLocaleInFront( context.path );
+
+		if ( pathWithoutLocale !== context.path ) {
+			return page.redirect( pathWithoutLocale );
+		}
+	}
+
+	next();
+};
+
```

Comment:
I just renamed and moved that method closer to `redirectWithoutLocaleParamIfLoggedIn()` as they are very similar, and it wasn't clear at first sight how they would differ.